bin/internal/shared.bat

```diff
@@ -92,7 +92,12 @@ GOTO :after_subroutine
     SET update_dart_bin=%FLUTTER_ROOT%/bin/internal/update_dart_sdk.ps1
     REM Escape apostrophes from the executable path
     SET "update_dart_bin=!update_dart_bin:'=''!"
-    %powershell_executable% -ExecutionPolicy Bypass -Command "Unblock-File -Path '%update_dart_bin%'; & '%update_dart_bin%'"
+    %powershell_executable% -ExecutionPolicy Bypass -Command "Unblock-File -Path '%update_dart_bin%'; & '%update_dart_bin%'; exit $LASTEXITCODE;"
```

Comment:
so `exit $LASTEXITCODE` is setting `%ERRORLEVEL%`?

Reply:
Yes, by default when you run powershell with -Command, all non-zero exit codes are translated into `1`.  This LASTEXITCODE trick allows me to set the exit code to the value from the command, which I set to 2 specifically for the case where the version is incorrect.

Reply:
Could you add a comment, just because I know I won't remember that? Otherwise, LGTM.

Reply:
Added more details and combined the two comments